river/sync.go

```diff
@@ -192,10 +192,21 @@ func (r *River) makeInsertReqData(req *elastic.BulkRequest, rule *Rule, values [
 	req.Action = elastic.ActionIndex
 
 	for i, c := range rule.TableInfo.Columns {
-		if name, ok := rule.FieldMapping[c.Name]; ok {
-			// has custom field mapping
-			req.Data[name] = r.makeReqColumnData(&c, values[i])
-		} else {
+		mapped := false
+		for _, m := range rule.FieldMapping {
+			if m.Mysql == c.Name {
+				mapped = true
+				v := r.makeReqColumnData(&c, values[i])
+				if m.Type == "list" {
```

Comment:
I think we should define a const type like TypeList instead of using  "list" directly.